codice/ddf
# DDF-1366 Re-implemented the SourcePoller

##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Links to 2.13.x PRs:
- https://github.com/codice/ddf/pull/3858
- https://github.com/codice/ddf/pull/4407
____

#### What does this PR do?
This PR ports over the changes for the new Source Poller implementation. Additionally, this PR includes a new commit to move the Source Poller code from core-api-impl to standard-framework because it is only used in the Standard Framework. It is not whitelisted.

_The file changes in this PR do not need to be re-reviewed, but this PR does need to be re-heroed._

#### Who is reviewing it? 
@jhunzik @Kjames5269 @lessarderic @mcalcote @pklinef @vinamartin

#### Select relevant component teams: 
@codice/core-apis 
@codice/io 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@clockard

#### How should this be tested?
See the test steps on https://github.com/codice/ddf/pull/3858 and https://github.com/codice/ddf/pull/4407. Additionally, confirm that the Source Poller does not cause any issues (searching, federating, log messages, source availability) when the DDF is a registry or is connected to a registry.
```log:set DEBUG org.codice.ddf.catalog.sourcepoller```

#### What are the relevant tickets?
[DDF-1366](https://codice.atlassian.net/browse/DDF-1366)

#### Screenshots
![image](https://user-images.githubusercontent.com/8041246/54237330-ebb77300-44d2-11e9-9b01-194019f86808.png)

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
